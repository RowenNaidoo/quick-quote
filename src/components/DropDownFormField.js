import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const DropDownFormField = ({ selectedValue, fieldId, headerText, onFieldChanged, errorMessage, options, isRequired, rightRadius = true }) => (
  <Conatiner>
    <Element>
      {headerText}{isRequired && <RequiredIndicator>*</RequiredIndicator>}
    </Element>
    <Element>
      <Field id={fieldId} value={selectedValue} rightRadius={rightRadius} onChange={(e) => { onFieldChanged(fieldId, e.target.value) }}>
        <option value=''>Select...</option>
        {
          options.map(option => <option key={option.value} value={option.value}>{option.text}</option>)
        }
      </Field>
    </Element>
    <Element>
      <ValidationMessage>
        {errorMessage}
      </ValidationMessage>
    </Element>
  </Conatiner>
)

const Conatiner = styled.div`
  display: flex;
  flex-direction: column;
`
const Element = styled.div`
  flex: 1 1 auto;
  justify-content: left;
`
const Field = styled.select`
  height: 36px;
  width: 100%;
  border-radius: 3px;
  border: 1px solid lightgray;
  padding: 0 10px;
  border-top-right-radius: ${props => props.rightRadius || '0px'};
  border-bottom-right-radius: ${props => props.rightRadius || '0px'};
`
const ValidationMessage = styled.span`
  color: red;
  font-size: 14px;
`
const RequiredIndicator = styled.span`
  color: red;
  font-size: 20px;
`

DropDownFormField.propTypes = {
  selectedValue: PropTypes.string,
  fieldId: PropTypes.string,
  headerText: PropTypes.string,
  onFieldChanged: PropTypes.func,
  errorMessage: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.object),
  isRequired: PropTypes.bool
}

export default DropDownFormField