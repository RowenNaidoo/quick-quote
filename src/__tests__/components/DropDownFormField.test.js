import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import DropDownFormField from '../../components/DropDownFormField'


describe('<DropDownFormField />', () => {
  let component
  let mockFieldChanged = jest.fn()

  const defaultProps = {
    selectedValue: 'USD',
    fieldId: 'ddFieldId',
    headerText: 'some header text',
    onFieldChanged: mockFieldChanged,
    errorMessage: '',
    options: [
      { value: 'AUD', text: 'Australian Dollar(AUD)' },
      { value: 'USD', text: 'US Dollar(USD)' }
    ],
    isRequired: true
  }

  beforeEach(() => {
    component = shallow(<DropDownFormField {...defaultProps} />)
  })

  test('matches snapshot', () => {
    component = renderer.create(<DropDownFormField {...defaultProps} />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})