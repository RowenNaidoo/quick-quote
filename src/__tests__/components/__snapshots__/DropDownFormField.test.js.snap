// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropDownFormField /> matches snapshot 1`] = `
<div
  className="sc-AxjAm iPShpD"
>
  <div
    className="sc-AxirZ cpGiat"
  >
    some header text
    <span
      className="sc-AxhUy enGJvm"
    >
      *
    </span>
  </div>
  <div
    className="sc-AxirZ cpGiat"
  >
    <select
      className="sc-AxiKw lcvSQm"
      id="ddFieldId"
      onChange={[Function]}
      value="USD"
    >
      <option
        value=""
      >
        Select...
      </option>
      <option
        value="AUD"
      >
        Australian Dollar(AUD)
      </option>
      <option
        value="USD"
      >
        US Dollar(USD)
      </option>
    </select>
  </div>
  <div
    className="sc-AxirZ cpGiat"
  >
    <span
      className="sc-AxhCb jNwDnw"
    >
      
    </span>
  </div>
</div>
`;
